src/Microsoft.IdentityModel.Tokens/Validation/SignatureValidationResult.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Microsoft.IdentityModel.JsonWebTokens.Results
+{
+#nullable enable
+    /// <summary>
+    /// Contains the result of validating a signature.
+    /// The <see cref="TokenValidationResult"/> contains a collection of <see cref="ValidationResult"/> for each step in the token validation.
+    /// </summary>
+    internal class SignatureValidationResult: ValidationResult
+    {
+        private Exception? _exception;
+
+        /// <summary>
+        /// Creates an instance of <see cref="SignatureValidationResult"/> representing the successful result of validating a signature.
+        /// </summary>
+        public SignatureValidationResult() : base(ValidationFailureType.ValidationSucceeded)
+        {
+            IsValid = true;
```

Comment:
By default IsValid == false?

Reply:
Yes, all of our results are initialised to not valid and not evaluated if nothing is changed.